src/main/kotlin/no/nav/familie/ef/sak/service/VurderingService.kt

```diff
@@ -1,35 +1,88 @@
 
+import no.nav.familie.ef.sak.api.Feil
 import no.nav.familie.ef.sak.api.dto.Aleneomsorg
 import no.nav.familie.ef.sak.api.dto.InngangsvilkårDto
-import no.nav.familie.ef.sak.integration.FamilieIntegrasjonerClient
+import no.nav.familie.ef.sak.api.dto.VurderingDto
 import no.nav.familie.ef.sak.integration.PdlClient
 import no.nav.familie.ef.sak.integration.dto.pdl.Familierelasjonsrolle
 import no.nav.familie.ef.sak.mapper.AleneomsorgMapper
 import no.nav.familie.ef.sak.mapper.MedlemskapMapper
-import no.nav.familie.ef.sak.repository.domain.SakMapper
+import no.nav.familie.ef.sak.repository.CustomRepository
+import no.nav.familie.ef.sak.repository.VilkårVurderingRepository
+import no.nav.familie.ef.sak.repository.domain.VilkårType
+import no.nav.familie.ef.sak.repository.domain.VilkårVurdering
+import no.nav.familie.ef.sak.repository.findByIdOrThrow
 import org.springframework.stereotype.Service
 import java.time.LocalDate
 import java.util.*
 
 @Service
-class VurderingService(private val sakService: SakService,
+class VurderingService(private val behandlingService: BehandlingService,
                        private val pdlClient: PdlClient,
+                       private val customRepository: CustomRepository,
+                       private val vilkårVurderingRepository: VilkårVurderingRepository,
                        private val medlemskapMapper: MedlemskapMapper) {
 
-    fun hentInngangsvilkår(sakId: UUID): InngangsvilkårDto {
-        val sak = sakService.hentOvergangsstønad(sakId)
-        val fnr = sak.søknad.personalia.verdi.fødselsnummer.verdi.verdi
+    fun oppdaterVilkår(vurdering: VurderingDto): UUID {
+        val vilkårVurdering = vilkårVurderingRepository.findByIdOrThrow(vurdering.id)
+
+        val behandlingId = vilkårVurdering.behandlingId
+        if (behandlingErLåstForVidereRedigering(behandlingId)) {
+            throw Feil(message = "Bruker prøver å oppdatere en vilkårsvurdering der behandling=$behandlingId er låst for videre redigering",
+                       frontendFeilmelding = "Behandlingen er låst for videre redigering")
+        }
+
+        val nyVilkårsVurdering = vilkårVurdering.copy(resultat = vurdering.resultat,
+                                                      begrunnelse = vurdering.begrunnelse,
+                                                      unntak = vurdering.unntak)
+        return vilkårVurderingRepository.save(nyVilkårsVurdering).id
+    }
+
+    fun hentInngangsvilkår(behandlingId: UUID): InngangsvilkårDto {
+        val søknad = behandlingService.hentOvergangsstønad(behandlingId)
+        val fnr = søknad.personalia.verdi.fødselsnummer.verdi.verdi
         val pdlSøker = pdlClient.hentSøker(fnr)
 
-        val medlemskap = medlemskapMapper.tilDto(medlemskapsdetaljer = sak.søknad.medlemskapsdetaljer.verdi,
+        val medlemskap = medlemskapMapper.tilDto(medlemskapsdetaljer = søknad.medlemskapsdetaljer.verdi,
                                                  pdlSøker = pdlSøker)
```

Comment:
Tar vi for gitt at vi alltid skal hente søknad om overgangsstønad? 
Det kan jo være en annen stønad eller en behandling uten en søknad? Eller bryr vi oss ikke om dette ennå?

Reply:
Det er noe vi burde se på når vi har en databasmodell for selve søknaden (hvis vi skal ha en), vi gjorde det enkelt nå ettersom då vi tenkte vi begynner med overgangsstønad

Reply:
For øvrig sjekket vi i kontrakten at medlemskapsdetaljer er med i alle 3 søknadstypene :)

Reply:
👌